File: guests/python/window.py
"""The window conformance scene, Python port — see guests/rust/window.rs
and tools/scenes/window.steps. The north-star spelling from DESIGN.md's
appendix, live: the surface's props ride the window() scope itself."""

import sys

import kaya

app = kaya.App()

with app.window(title="window probe", width=640, height=400):
    probe = kaya.signal("window probe")
    with kaya.column():
        kaya.label(bind=probe)  # label#0

sys.exit(app.run())

File: guests/js/window.ts
// The window conformance scene (tools/scenes/window.steps).

import * as kaya from "kaya-gui";

const app = new kaya.App();

app.window({ title: "window probe", width: 640, height: 400 }, () => {
  const probe = kaya.signal("window probe");
  kaya.column(() => {
    kaya.label({ bind: probe }); // label#0
  });
});

app.run();
